clients/python-wrapper/lakefs/object.py

```diff
@@ -568,10 +570,17 @@ def _upload_presign(self) -> lakefs_sdk.ObjectStats:
                                            checksum=etag,
                                            user_metadata=self.metadata,
                                            content_type=self.content_type)
-        return self._client.sdk_client.staging_api.link_physical_address(self._obj.repo,
-                                                                         self._obj.ref,
-                                                                         self._obj.path,
-                                                                         staging_metadata=staging_metadata)
+        if_none_match = "*" if self._mode.startswith("x") else None
+        try:
+            return self._client.sdk_client.staging_api.link_physical_address(self._obj.repo,
+                                                                             self._obj.ref,
+                                                                             self._obj.path,
+                                                                             staging_metadata=staging_metadata,
+                                                                             if_none_match=if_none_match)
+        except lakefs_sdk.ApiException as e:
+            if self._mode.startswith("x") and e.status == http.HTTPStatus.PRECONDITION_FAILED:
```

Comment:
would check if_none_match is not none / set instead of use the same compare logic again.